docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,598 @@
+
+If you've come this far, then the "backend" is done. You can now run `gatsby build` to see your indices in Algolia's webinterface flooded with all your data (provided the build throws no errors).
+
+## Frontend
+
+What remains to be done now is to add a user-facing search interface to your site. It needs a way for the user to enter a search string, send that string to Algolia, receive matching results from your indices and finally display those to the user. Let's dive right in.
+
+We're going to assemble everything we need into a React `Search` component that we call from anywhere on our site where we want the user to be able to search. Even though design varies strongly from site to site, we'll also go through the styles in this guide since working out the CSS transitions to have the search field slide out as the user clicks on it and the results pane to appear once Algolia returns some matches can take some time. We'll implement this functionality with `styled-components` but of course, you can use any styling method you like and the CSS should remain mostly the same.
+
+The first step is to create the main component file.
+
+```jsx
+// src/components/Search/index.js
+import React, { Component, createRef } from 'react'
+import {
+  InstantSearch,
+  Index,
+  Hits,
+  connectStateResults,
+} from 'react-instantsearch-dom'
+import { Algolia } from 'styled-icons/fa-brands/Algolia'
+
+import { Root, HitsWrapper, By } from './styles'
+import Input from './Input'
+import * as hitComps from './hits'
+
+const events = ['mousedown', 'touchstart']
+
+const Results = connectStateResults(
+  ({ searchState: state, searchResults: res, children }) =>
+    res && res.nbHits ? children : `No results for ${state.query}`
+)
+
+const Stats = connectStateResults(
+  ({ searchResults: res }) =>
+    res && res.nbHits > 0 && `${res.nbHits} result${res.nbHits > 1 ? `s` : ``}`
+)
+
+export default class Search extends Component {
+  state = { query: ``, focussed: false, ref: createRef() }
+
+  updateState = state => this.setState(state)
+
+  focus = () => {
+    this.setState({ focussed: true })
+  }
+
+  disableHits = () => {
+    this.setState({ focussed: false })
+  }
+
+  handleClickOutside = event => {
+    if (!this.state.ref.current.contains(event.target)) {
+      this.setState({ focussed: false })
+    }
+  }
+
+  componentDidMount() {
+    events.forEach(event =>
+      document.addEventListener(event, this.handleClickOutside)
+    )
+  }
+
+  componentWillUnmount() {
+    events.forEach(event =>
+      document.removeEventListener(event, this.handleClickOutside)
+    )
+  }
+
+  render() {
+    const { query, focussed, ref } = this.state
+    const { indices, collapse, hitsAsGrid } = this.props
+    return (
+      <InstantSearch
+        appId={process.env.GATSBY_ALGOLIA_APP_ID}
+        apiKey={process.env.GATSBY_ALGOLIA_SEARCH_KEY}
```

Comment:
would use the `searchClient` form (where you call `algoliasearch` yourself), so we don't need to update this guide later for that

Reply:
Now calling `algoliasearch` and passing the returned `searchClient` to `<InstantSearch />`.